### PR TITLE
update which to be compatible with python3.4

### PR DESCRIPTION
**Pull request by gforsyth**
and one other, small change

this should address #868

### CODE REVIEW

File: xonsh/aliases.py [hunk 1]
```diff
@@ -329,11 +329,25 @@ def bang_bang(args, stdin=None):
...
+                                                 stderr=subprocess.PIPE)):
             return '<no version number available on OS X>'
 
-    _ver = subprocess.run(['which','-v'], stdout=subprocess.PIPE)
-    return(_ver.stdout.decode('utf-8'))
+    _ver = subprocess.check_output(['which','-v'])
+    return _ver.decode('UTF-8')
+
+
+class AWitchAWitch(Action):
```
Comment (scopatz) [inline xonsh/aliases.py]: <3

File: xonsh/aliases.py [hunk 2]
```diff
@@ -361,16 +376,20 @@ def which(args, stdin=None):
...
         if pargs.all:
             try:
-                subprocess.run(['which'] + args,
-                               stderr=subprocess.PIPE,
-                               check=True)
+                subprocess.check_call(['which'] + args,
+                                      stdout=subprocess.PIPE,
+                                      stderr=subprocess.PIPE)
+                subprocess.call(['which'] + args,
+                               stderr=subprocess.PIPE)
```
Comment (scopatz) [inline xonsh/aliases.py]: I don't grok why you have these two calls.  Would you mind adding a comment explaining it? Thanks!

Comment (gforsyth) [inline xonsh/aliases.py]: The reason you don't grok it is because they're redundant. Fix incoming.

**Comment by gforsyth** [comment]
Ready for another look @scopatz

**Comment by scopatz** [comment]
Thanks a ton @gforsyth!

**Pull request merged**